src/components/frame/v5/pages/AgreementsPage/partials/AgreementCard/AgreementCard.tsx

```diff
@@ -0,0 +1,236 @@
+import UserPopover from '~v5/shared/UserPopover/UserPopover.tsx';
+
+import { type AgreementCardProps } from './types.ts';
+import { renderAgreementCardSkeleton } from './utils.tsx';
+
+const AgreementCard: FC<AgreementCardProps> = ({ transactionId }) => {
+  const {
+    action,
+    refetchMotionState,
+    loadingAction,
+    motionState,
+    startPollingForAction,
+    stopPollingForAction,
+  } = useGetColonyAction(transactionId);
+  const { decisionData, motionData } = action || {};
+  const {
+    createdAt: date = '',
+    description = '',
+    title,
+    motionDomainId: team,
+    walletAddress = '',
+  } = decisionData || {};
+  const { motionId = '', motionStakes, requiredStake = '' } = motionData || {};
+  const navigate = useNavigate();
+  const { colony } = useColonyContext();
+  const { user, loading } = useUserByNameOrAddress(walletAddress);
+  const currentTeam = colony?.domains?.items.find(
+    (domain) => domain?.nativeId === team,
+  );
+
+  const networkMotionStateEnum = getEnumValueFromKey(
+    NetworkMotionState,
+    motionState,
+    0,
+  );
+
+  const [activeStepKey, setActiveStepKey] = useState<Steps>(
+    networkMotionStateEnum,
+  );
+
+  const motionFinished =
+    motionState === NetworkMotionState.Finalizable ||
+    motionState === NetworkMotionState.Finalized ||
+    motionState === NetworkMotionState.Failed;
+
+  useEffect(() => {
+    startPollingForAction(getSafePollingInterval());
+    setActiveStepKey(networkMotionStateEnum);
+    if (motionFinished) {
+      setActiveStepKey(CustomStep.Finalize);
+    }
+    return () => stopPollingForAction();
+  }, [
+    motionFinished,
+    networkMotionStateEnum,
+    startPollingForAction,
+    stopPollingForAction,
+  ]);
+
+  const motionStateEnum: MotionState | undefined = useMemo(() => {
+    if (!motionData) return undefined;
+    const motionStakesRaw = motionData?.motionStakes?.raw;
+    const revealedVotesPercentage = motionData?.revealedVotes.percentage || '';
+
+    if (
+      activeStepKey === NetworkMotionState.Finalizable &&
+      BigNumber.from(motionStakesRaw?.nay).gte(requiredStake) &&
+      BigNumber.from(motionStakesRaw?.yay).gte(requiredStake)
+    ) {
+      if (
+        BigNumber.from(revealedVotesPercentage?.yay).gt(
+          revealedVotesPercentage?.nay,
+        )
+      ) {
+        return MotionState.Passed;
+      }
+
+      return MotionState.Failed;
+    }
+
+    return motionData
+      ? getMotionState(networkMotionStateEnum, motionData)
+      : MotionState.Staking;
+  }, [activeStepKey, motionData, networkMotionStateEnum, requiredStake]);
+
+  return (
+    <div className="w-full h-full flex flex-col py-6 px-5 rounded-lg border border-gray-200">
+      {loadingAction ? (
+        renderAgreementCardSkeleton()
+      ) : (
+        <>
+          <div className="flex items-center justify-between mb-4">
+            {motionStateEnum && <MotionStateBadge state={motionStateEnum} />}
+            {motionStakes &&
```

Comment:
could we wrap these 3 Boolean statements into 1 variable called something like `isMotionActive`